### DISCUSSION TITLE
Two questions coming from gqlgen to better understand if my life is about to improve :smile:

### DISCUSSION THREAD

**Discussion by frederikhors** [async-graphql/async-graphql#929] [General]
I need to evaluate `async-graphql` for a new big project.

We'are using Golang & gqlgen right now.

I have two questions for which I have no answer from a first quick lap:

1. Can I detect which fields are requested to be returned? So that I can exclude/include the DB's columns in the queries?

    Example: if the user only requests `id` and `name` without `age` and `score`, like:
    
    ```graphql
    player {
      id
      name
      // age
      // score
    }
    ```

    can I detect that only those two fields (`id` and `name`) have been requested and that the others have been ignored?

1. Can I detect in mutation resolvers if the user has deliberately chosen `null` as the value of nullable fields (string/int/time/...) or if he is using sparse updates and has omitted those fields entirely?

    [In gqlgen this is a big mess](https://github.com/99designs/gqlgen/issues/1416).

**Comment by sunli829** [comment]
> 1. Can I detect which fields are requested to be returned? So that I can exclude/include the DB's columns in the queries?

I guess you want to use it to optimize execution performance?

> 2. Can I detect in mutation resolvers if the user has deliberately chosen null as the value of nullable fields (string/int/time/...) or if he is using sparse updates and has omitted those fields entirely?

You can use `MaybeUndefined<T>`. 🙂

**Reply by frederikhors** [reply]
> I guess you want to use it to optimize execution performance?

Yep. Is it possible?

**Reply by sunli829** [reply]
Of course, you can do it like this:

https://docs.rs/async-graphql/4.0.0/async_graphql/context/struct.ContextBase.html#method.look_ahead
